lib/src/search_page.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:flutter/material.dart';
+
+import '../consts.dart';
+import 'countries.dart';
+
+class SearchPage extends StatefulWidget {
+  const SearchPage({super.key});
+
+  @override
+  State<SearchPage> createState() => _SearchPageState();
+}
+
+class _SearchPageState extends State<SearchPage> {
+  List<Map<String, String>> allCountries = Countries.allCountries;
+  List<Map<String, String>> filteredCountries = [];
+  void initState() {
+    super.initState();
+    filteredCountries = allCountries; // Initialize with all countries
+  }
+
+  void updateFilteredCountries(List<Map<String, String>> countries) {
+    setState(() {
+      filteredCountries = countries;
+    });
+  }
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text("Select a Country"),
+      ),
+      body: Column(
+        children: [
+          CountrySearchBar(
+            allCountries: allCountries,
+            onSearch: updateFilteredCountries,
+          ),
+          Expanded(
+            child: ListView.builder(
+              itemCount: filteredCountries.length,
+              itemBuilder: (context, index) {
```

Comment:
You can probably add dividers to each item in the list so to easily read the values.

Reply:
how is the spacing and divider

Reply:
<img width="320" alt="Screenshot 2023-10-03 at 11 50 53 PM" src="https://github.com/gopavasanth/WhatsApp-Karo/assets/30266744/31abb5a9-08fc-42e4-a55d-e9ef43016882">

Make sure these don't overflow

Reply:
<img width="303" alt="Screenshot 2023-10-03 at 11 56 43 PM" src="https://github.com/gopavasanth/WhatsApp-Karo/assets/30266744/75e3fbf6-62e6-44c0-bb7b-3e5680be39d2">

Also, handle them properly when font size is increased (display settings in mobile)

Also, make the divider color a bit light, it feels too dark

Reply:
sure on it

Reply:
Can you add some screenshots @Mrudul111? I am currently outside and will only be able to look at this in the night.

Reply:
sure

Reply:
![image](https://github.com/gopavasanth/WhatsApp-Karo/assets/90367225/8ed762c1-3889-4e36-9bf8-d32b7dd7d837)

Reply:
![image](https://github.com/gopavasanth/WhatsApp-Karo/assets/90367225/48803c6c-2640-4f27-9225-99a2cecb2c1f)

Reply:
sorry for the delay man but all the issues are fixed now